test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -125,6 +129,16 @@ spec =
             Meta.is_a Date_Time.now.zone Time_Zone . should_be_true
             Meta.is_a Date_Time.now.zone Date . should_be_false
 
+        Test.specify "constructor variants check" <|
+            (Sum_Type.Variant_A 42).is_a Sum_Type . should_be_true
+            (Sum_Type.Variant_A 42).is_a Sum_Type.Variant_A . should_be_false
+            (Sum_Type.Variant_A 42).is_a Sum_Type.Variant_B . should_be_false
```

Comment:
Shouldn't this instead throw `Illegal_Argument` error as we [discsussed in the issue](https://github.com/enso-org/enso/issues/5792#issuecomment-1453568144)?

Reply:
Up to you and @jdunkerley. I don't like throwing needless exceptions in production - as [Martin Rinard's presentation](http://wiki.apidesign.org/images/1/11/RinardOOPSLA06.pdf) suggests - letting the system run is better than prematurely crashing it. As such I am completely fine with `False` return value - not only it **doesn't crash** the system, it is actually also **correct** in this case.